### PR TITLE
XW-3639 | Logger for JW

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js [hunk 1]
```diff
@@ -90,7 +94,7 @@ define('wikia.articleVideo.featuredVideo.events', function () {
 		playerInstance.on('play', function (data) {
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js]: MAJOR Remove the unused function parameter "data". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	function info(name, description) {
+		console.info(prefix, name, description);
+	}
+
+	function warn(name, description) {
+		console.warn(prefix, name, description);
+	}
+
+	function error(name, description) {
+		logErrorToService(prefix, name, description);
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js]: CRITICAL "logErrorToService" declared at line 5 expects 2 arguments, but 3 were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/9455424201eb6b4cbc9340c4789fddff03411e89/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/fe5d86fde84cd8f5c9ff4419843304cdd77bfe43/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/ef463edb484066514d2825273fe033660a1d9f7f/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/2e682cdde9e548b73894499488425e9ffea5525c/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/a0f8bd82225fafa4700ca3c809d0cdbb7634f858/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.instant.js [hunk 3]
```diff
@@ -78,12 +80,14 @@ require([
...
 		});
-		console.info('jwplayer after setup');
+		logger.info('jwplayer after setup');
 
 		featuredVideoAds(playerInstance, bidParams);
 		featuredVideoEvents(playerInstance, willAutoplay);
 		featuredVideoTracking(playerInstance, willAutoplay);
 		featuredVideoMoatTracking(playerInstance);
+		logger.subscribeToPlayerSetupError(playerInstance);
+		logger.subscribeToInternalPlayerErrors(playerInstance);
```
Comment (xkxd) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.instant.js]: i'd move this to line 84 before any of submodules is called.
\+ maybe combine them in `subscribeToPlayerErrors`?

**Review by xkxd — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/4058f6486a2868cbfaa3b6143b91a1cacbf79fd4/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	function info(name, description) {
+		console.info(prefix, name, description);
+	}
+
+	function warn(name, description) {
+		console.warn(prefix, name, description);
+	}
+
+	function error(name, description) {
+		logErrorToService(name, description);
```
Comment (latata) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js]: I'd add console.error here

**Review by latata — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/e5b4b68f9733baa2e3611609c35136b95065960a/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/ef025720680c6b1c9f33bb981d0134bd8922db45/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [featured-video.jwplayer.events.js#L35](https://github.com/Wikia/app/blob/88bd8a085aafd6b9abaf9aaeae55b6794a970948/extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.events.js#L35): Function has a complexity of 11 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by xkxd — APPROVED** [review]

File: extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+			};
+
+		if (description) {
+			data.description = typeof description === 'string' ? description : JSON.stringify(description);
+		}
+
+		request.open('POST', loggerUrl, true);
+		request.setRequestHeader('Content-type', 'application/json');
+
+		request.send(JSON.stringify(data));
```
Comment (xkxd) [inline extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js:18]: since we do `stringify` here, does it still necessary in L#12?

Comment (latata) [reply extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.logger.js:18]: Backend needs string as a description

**Pull request merged**